src/db/sysdb_services.c

```diff
@@ -151,8 +151,7 @@ sysdb_getservbyport(TALLOC_CTX *mem_ctx,
         res->msgs = talloc_steal(res, msgs);
     }
 
-    *_res = res;
-
+    *_res = talloc_move(mem_ctx, &res);
```

Comment:
Are you using `talloc_move()` instead of `talloc_steal()` in case `ret != EOK` and thus `res == NULL`?

Reply:
Not specially for that reason, it is just a personal preference. If `ret != EOK` then `res` is `NUL` and there is no difference between `talloc_steal()` and `talloc_move()`.